catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -258,8 +274,8 @@ private boolean orderContainsSupportedDelivery() {
         if (order.del_list != null) {
             for (DeliveryDetails deliveryDetails : order.del_list) {
                 Destination destination = deliveryDetails.dests;
-                if (destination.discriminator() == DestinationType.FTP
-                        && destination.f_dest() != null) {
+                if (((destination.discriminator() == DestinationType.FTP) && (destination.f_dest() != null)) ||
+                        ((destination.discriminator() == DestinationType.EMAIL) && (destination.e_dest() != null))) {
```

Comment:
It's been pointed out that I completely misread the OR condition connecting the two subs. This would work, but isn't radically cleaner than a cleaned up procedural solution:

```java
    private static final Predicate<Destination> FTP_PRED =
            d -> d.discriminator() == DestinationType.FTP && d.f_dest() != null;

    private static final Predicate<Destination> EMAIL_PRED =
            d -> d.discriminator() == DestinationType.EMAIL && d.e_dest() != null;
// ...
        Arrays.stream(order.del_list)
                .map(d -> d.dests)
                .filter(FTP_PRED.or(EMAIL_PRED))
                .anyMatch(Objects::nonNull);
```

You could also create multiple predicates and `and` them together instead of using explicit `&&` conditions, but that gets appallingly weird.

There is something to be said for creating your Predicates one time, statically, and then reusing them. You get the best ROI on that when you want to reuse your Predicates for other conditional tests in your class.